src/main/java/org/jenkinsci/plugins/credentialsbinding/Binding.java

```diff
@@ -77,7 +79,7 @@ public SingleEnvironment(String value, Unbinder unbinder) {
     }
 
     @Deprecated
-    public interface Environment {
+    public interface Environment extends Serializable {
```

Comment:
You just need to suppress that warning. `Environment` is deprecated for a reason: it was not, in general, serializable. `UnbinderWrapper` is used only if a `Binding` subtype failed to be updated to support Pipeline, still implementing `bind`, and such implementations would typically have an unserializable environment.